icc-x-api/icc-crypto-x-api.ts

```diff
@@ -1358,39 +1347,154 @@ export class IccCryptoXApi {
     }
   }
 
-  saveKeyChainInHCPFromLocalStorage(hcpId: string): Promise<HealthcarePartyDto> {
-    return this.hcpartyBaseApi
+  /**
+   * Save eHealth certificate and validity in the encrypted fields of the HCP.
+   * @param hcpId Id of the hcp to modify
+   * @param autoupdateHCP update the HCP using the base HCP service; if false simply return the new HCP,
+   * without publishing the modifications on the db (this way this can be handled elsewhere e.g. by the
+   * HCP x api service, which makes use of caching).
+   */
+  async saveKeyChainInHCPFromLocalStorage(
+    hcpId: string,
+    autoupdateHCP = true
+  ): Promise<HealthcarePartyDto> {
+    return await this.hcpartyBaseApi
       .getHealthcareParty(hcpId)
-      .then(hcp => {
+      .then(async (hcp: HealthcarePartyDto) => {
+        let aesKey: CryptoKey | null = null
+        try {
+          aesKey = _.find(
+            await this.decryptAndImportAesHcPartyKeysForDelegators([hcp.id!], hcp.id!),
+            delegator => delegator.delegatorId === hcp.id
+          )!.key
+        } catch (e) {
+          console.error("Error while importing the AES key.")
+        }
+        if (!aesKey) {
+          console.error("No encryption key!")
+        }
+
         const opts = hcp.options || {}
 
         const crt = this.getKeychainInBrowserLocalStorageAsBase64(hcp.id!!)
         _.set(opts, this.hcpPreferenceKeyEhealthCert, crt)
```

Comment:
The unencrypted data is still stored here. Instead we should clear it if present.

Reply:
https://medispring.atlassian.net/browse/MS-1899 it was determined during the tech analysis to keep the unencrypted data fields for a while, in order to avoid making a global migration, or to avoid blocking the users who who don't have their eHealth certificate with them at the time of deployment of this feature.

The unencrypted keys will be removed in the future, one we're sure all HCPs (or most of them) will have encrypted fields.

Reply:
After discussing it with Phil, we decided to go with your idea: removing any present unencrypted certificates and pushing only the encrypted ones to the back-end.

Reply:
Done!